jvb/src/main/kotlin/org/jitsi/videobridge/cc/allocation/ReceiverConstraintsMap.kt

```diff
@@ -39,9 +39,22 @@ class ReceiverConstraintsMap {
         synchronized(lock) {
             return map.remove(key)?.also { removed ->
                 if (removed.maxHeight == maxHeight) {
-                    maxHeight = map.values.map { it.maxHeight }.max() ?: 0
+                    maxHeight = findNewMax(maxHeight)
                 }
             }
         }
     }
+
+    // Note: this method must be called with the lock held
+    private fun findNewMax(oldMax: Int): Int {
```

Comment:
Maybe name this so it's clear it's only applicable for `remove`?

Reply:
Well, technically it doesn't care when it's used.  I would've done it in `remove` itself but with the early returns it's tricky.

Reply:
Well, it's not correct for `add`, but fortunately `add`'s calculation is easy.

Reply:
Yeah, actually you're right.  Will rename.